Add documentation about Goblint's transfer functions (#1053)
goblint/analyzer: docs/developer-guide/firstanalysis.md

The assignment relies on the function `eval`, which is almost there. It just needs you to fix the evaluation of constants! Unless you jumped straight to this line, it should not be too complicated to fix this.
With this in place, we should have sufficient information to tell Goblint that the assertion does hold.

> For more information on the signature of the individual transfer functions, please check out their documentation in the file which they're defined in: [src/framework/analyses.ml](https://github.com/goblint/analyzer/blob/master/src/framework/analyses.ml#LL355C1-L355C17).

Reviewer: ```suggestion
For more information on the signature of the individual transfer functions, please check out `module type Spec` documentation in [`src/framework/analyses.ml`](https://github.com/goblint/analyzer/blob/master/src/framework/analyses.ml).
```
Linking the lines is bad because any changes to the file will shift things around and make them out of sync. And it's not feasible to remember to update this tutorial documentation every time someone makes a change in `Analyses`.

Author: Agreed. Thanks for the tip!
